### ISSUE TITLE
AUC as loss function

### ISSUE THREAD

**Issue by WojciechMigda** [cair/TsetlinMachine#10]
Hello,
I would be interested in having Tsetlin Machine optimizing for AUC. Do you maybe have some hints if this is plausible and if so share some pointers?
Thanks!

**Comment by olegranmo** [comment]
Hi @WojciechMigda! Did you have any success here? I recently experimented with weighting Type I and Type II feedback differently for each category. The purpose was to balance false positive against false negatives. Maybe that could be a start? Best wishes, Ole-Christoffer

**Comment by WojciechMigda** [comment]
Hi @olegranmo ,
no I did not pursue this problem any further, as it was related to a data science competition back then in which I tried to utilize Tsetlin Machine in and that ended not long after.
I have another question though, which stemmed from another competition in which I participated very recently. The problem to solve had a very impalanced dataset (about 1:50 ratio between two classes). In retrospect I think that if I applied some form of boosting then Tsetlin Machine would perform better. Did you experiment with boosting for TM? What are your experiences with TM vs. imbalanced data in general?
